src/app/tasks/task.service.ts

```diff
@@ -1,84 +1,87 @@
-import { Task, Option, BasicTask } from "./task.model";
-import { IdGeneratorService } from "./id-generator-service";
+import { Task, Option, BasicTask, TaskData } from "./task.model";
 import { Injectable } from "@angular/core";
 import { HttpClient } from "@angular/common/http";
 import { beApiRoutes } from "../routes/be-api.routes";
-import { BehaviorSubject } from 'rxjs';
+import { BehaviorSubject, first, Observable, tap } from 'rxjs';
 
 @Injectable()
-export class TaskService {
+export class TaskService{
     
-    public taskTypesSubject = new BehaviorSubject<Option[]>([]);
-    public taskStatusesSubject = new BehaviorSubject<Option[]>([]);
-    public tasksSubject = new BehaviorSubject<BasicTask[]>([]);
+    private taskTypes : Option[] = [];
+    private taskStatuses : Option[] = [];
+    public basicTasksSubject = new BehaviorSubject<BasicTask[]>([]);
+    public basicTasksObservable$ = this.basicTasksSubject.asObservable();
 
-    constructor(private idGeneratorService: IdGeneratorService, private httpClient: HttpClient) {
-        this.fetchDataFromDB();
+    constructor(private httpClient: HttpClient) {
+        this.getEssentialData();
```

Comment:
does this fetch data? you dont seem to subscribe to it anywhere so I think it does not fetch anything

Reply:
Removed the call in the service constructor but the task list component subscribes to getEssentialData()